src/rdb_protocol/terms/sort.cc

```diff
@@ -83,9 +83,9 @@ class orderby_term_t : public op_term_t {
         /* Add a sorting to the table if we're doing indexed sorting. */
         } else if (index.has()) {
             rcheck(tbl_slice.has(), base_exc_t::LOGIC,
-                   "Indexed order_by can only be performed on a TABLE or TABLE_SLICE.");
+                   "Indexed order_by can only be performed on a TABLE or TABLE_SLICE, not on a `NAME_OF_TYPE_ORDER_BY_WAS_CALLED_ON`.  Make sure ORDER_BY comes before any transformations such as MAP or FILTER.");
```

Comment:
I think it would be a bit more correct to say "transformations (such as `map`) or filters.", since it's debatable whether a `filter` is considered a transformation or not.